metrics-sender/src/main/java/io/knotx/metrics/graphite/GraphiteReporter.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (C) 2018, Coda Hale, Yammer Inc.
+ * Copyright (C) 2018 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * This code was partially copied from metrics-graphite project by Coda Hale, Yammer Inc.
+ * The copied code:
+ *      GraphiteReporter.Builder class and its methods
+ * The new code:
+ *      GraphiteReporter's overridden and new methods
+ */
+package io.knotx.metrics.graphite;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import com.codahale.metrics.Clock;
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricAttribute;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.codahale.metrics.graphite.Graphite;
+import com.codahale.metrics.graphite.GraphiteSender;
+
+public class GraphiteReporter extends com.codahale.metrics.graphite.GraphiteReporter {
```

Comment:
Could you rename this reporter to `LowNetworkConsumptionGraphiteReporter` or something that better describes what is the purpose of this class?

Reply:
renamed